### PR TITLE
S05 us03 dashboard candidate

### PR DESCRIPTION
**Pull request by axeljdev**

### CODE REVIEW

File: client/src/components/atomic/card/CardOfferForCandidate.jsx [hunk 1]
```diff
@@ -1,20 +1,56 @@
...
 import ScrollToTop from "../../../services/scrollToTop";
+import { AuthContext } from "../../../contexts/AuthContext";
 
 export default function CardOfferForCandidate({ offer }) {
+  const { auth } = useContext(AuthContext);
+  const [isFavorite, setIsFavorite] = useState(offer.is_favorite);
+
+const handleCheckboxChange = async (e) => {
+    const isChecked = e.target.checked;
+    const url = "http://localhost:3310/api/favorites";
```
Comment (adrale) [inline client/src/components/atomic/card/CardOfferForCandidate.jsx:15]: change by .env

File: client/src/components/atomic/card/CardOfferForCandidate.jsx [hunk 2]
```diff
@@ -1,20 +1,56 @@
...
+          method: "DELETE",
+          headers,
+          body,
+        });
+      }
+
+      if (!response.ok) throw new Error(`HTTP error! status: ${response.status}`);
+      setIsFavorite(isChecked);
+    } catch (err) {
+      console.error("Error handling favorite", err);
```
Comment (adrale) [inline client/src/components/atomic/card/CardOfferForCandidate.jsx:40]: console statement

File: client/src/main.jsx [hunk 3]
```diff
@@ -63,12 +64,7 @@ const router = createBrowserRouter([
       },
       {
         path: "/dashboardCandidate/:id",
-        element: (
```
Comment (adrale) [inline client/src/main.jsx:66]: vas-y remet

File: client/src/pages/DashboardCandidate.jsx [hunk 4]
```diff
@@ -1,10 +1,129 @@
...
-  return <h1>{lastname}</h1>;
+import "../../index.css";
+
+function DashboardCandidate() {
+  const data = useLoaderData();
+  const [favorites, setFavorites] = useState([]);
+  useEffect(() => {
+    const fetchFavorites = async () => {
+      try {
+        const response = await fetch("http://localhost:3310/api/favorites", {
```
Comment (adrale) [inline client/src/pages/DashboardCandidate.jsx:22]: t'as rechangé?

**Review by adrale — APPROVED** [review]

**Review by JulienChevreauPro — APPROVED** [review]

**Review by adrale — APPROVED** [review]

**Review by Yoann-Fortin — APPROVED** [review]

**Pull request merged**